packages/scaffold-ui/src/partials/w3m-connector-list/index.ts

```diff
@@ -49,63 +49,84 @@ export class W3mConnectorList extends LitElement {
       ConnectorUtil.getConnectorsByType(this.connectors, this.recommended, this.featured)
-    const isConnectedWithWC = ConnectorUtil.getIsConnectedWithWC()
-    const isWCEnabled = OptionsController.state.enableWalletConnect
 
-    return html`
-      <wui-flex flexDirection="column" gap="xs">
-        ${isWCEnabled && !isConnectedWithWC
-          ? html`<w3m-connect-walletconnect-widget
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-walletconnect-widget>`
-          : null}
-        ${recent.length
-          ? html`<w3m-connect-recent-widget
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-recent-widget>`
-          : null}
-        ${multiChain.length
-          ? html`<w3m-connect-multi-chain-widget
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-multi-chain-widget>`
-          : null}
-        ${announced.length
-          ? html`<w3m-connect-announced-widget
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-announced-widget>`
-          : null}
-        ${injected.length
-          ? html`<w3m-connect-injected-widget
-              .connectors=${injected}
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-injected-widget>`
-          : null}
-        ${featured.length
-          ? html`<w3m-connect-featured-widget
-              .wallets=${featured}
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-featured-widget>`
-          : null}
-        ${custom?.length
-          ? html`<w3m-connect-custom-widget
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-custom-widget>`
-          : null}
-        ${external.length
-          ? html`<w3m-connect-external-widget
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-external-widget>`
-          : null}
-        ${recommended.length
-          ? html`<w3m-connect-recommended-widget
-              .wallets=${recommended}
-              tabIdx=${ifDefined(this.tabIdx)}
-            ></w3m-connect-recommended-widget>`
-          : null}
-      </wui-flex>
-    `
+    const connectorPosition = ConnectorUtil.getConnectorPosition({
+      custom,
+      recent,
+      announced,
+      injected,
+      multiChain,
+      recommended,
+      featured,
+      external
+    })
+
+    return connectorPosition.map(position => {
+      switch (position) {
+        case 'walletConnect':
+          return html`<w3m-connect-walletconnect-widget
+            tabIdx=${ifDefined(this.tabIdx)}
+          ></w3m-connect-walletconnect-widget>`
+
+        case 'recent':
+          return html`<w3m-connect-recent-widget
+            tabIdx=${ifDefined(this.tabIdx)}
+          ></w3m-connect-recent-widget>`
+
+        case 'multiChain':
+          return html`<w3m-connect-multi-chain-widget
+            tabIdx=${ifDefined(this.tabIdx)}
+          ></w3m-connect-multi-chain-widget>`
```

Comment:
this is a tricky case cause we compute it ourselves. I think we should list these always above `injected` / `announced`